Mathlib/LinearAlgebra/AffineSpace/AffineSubspace/Defs.lean

```diff
@@ -858,6 +865,13 @@ theorem bot_lt_affineSpan : ⊥ < affineSpan k s ↔ s.Nonempty := by
   rw [bot_lt_iff_ne_bot, nonempty_iff_ne_empty]
   exact (affineSpan_eq_bot _).not
 
+lemma affineSpan_eq_top_iff_nonempty_of_subsingleton [Subsingleton P] :
+    affineSpan k s = ⊤ ↔ s.Nonempty := by
+  rw [← affineSpan_nonempty k, nonempty_iff_ne_bot]
+  rcases (affineSpan k s).eq_bot_or_eq_top_of_subsingleton with h | h
+  · simp [h]
+  · simp [h]
```

Comment:
I woudl geuss there's a shorter proof using `bot_lt_affineSpan`

Reply:
Shortened the proof.